R/datasets.R

```diff
@@ -0,0 +1,627 @@
+#' @description
+#' Get a registered Dataset from the workspace by its registration name.
+#'
+#' @param dataset The Dataset object.
+#' @param workspace The existing AzureML workspace in which the Dataset was registered.
+#' @param name The registration name.
+#' @param version The registration version. Defaults to "latest".
+#' @return The registered Dataset object.
+#' @export
+#' @md
+get_dataset_by_name <- function(workspace, name, version = "latest") {
+  azureml$data$abstract_dataset$AbstractDataset$get_by_name(workspace,
+                                                            name,
+                                                            version)
+}
+
+#' Get Dataset by ID.
+#'
+#' @description
+#' Get a Dataset which is saved to the workspace using its ID.
+#'
+#' @param workspace The existing AzureML workspace in which the Dataset is saved.
+#' @param id The ID of the dataset
+#' @return The Dataset object
+#' @export
+#' @md
+get_dataset_by_id <- function(workspace, id) {
+  azureml$data$abstract_dataset$AbstractDataset$get_by_id(workspace, id)
+}
+
+#' Return the named list for input datasets.
+#'
+#' @description
+#' Return the named list for input datasets.
+#'
+#' @param name The name of the input dataset
+#' @param run The run taking the dataset as input
+#' @return A dictionary with the name being the dataset's name and value being
+#' the delivered data. If the mode is set to mount or download, it will return
+#' the base path of the delivered data. If the mode is set to direct, it will
+#' return the dataset object.
+#' @export
+#' @md
+get_input_dataset_from_run <- function(name, run) {
+  run$input_datasets(name)
+}
+
+#' Create an unregistered, in-memory Dataset from files.
+#'
+#' @description
+#' Create an unregistered, in-memory Dataset from files. Use this method to read
+#' files as streams of data. Returns one file stream object per file read.
+#' The returned Dataset is not registered with the workspace.
+#'
+#' @param path A data path in a registered datastore or a local path.
+#' @param validate Indicates whether to validate if data can be loaded from the
+#' returned dataset. Defaults to True. Validation requires that the data source
+#' is accessible from the current compute.
+#' @return The Dataset object
+create_file_dataset_from_files <- function(path, validate = TRUE) {
+  azureml$data$dataset_factory$FileDatasetFactory$from_files(path, validate)
+}
+
+#' Get a list of file paths for each file stream defined by the dataset.
+#'
+#' @description
+#' Get a list of file paths for each file stream defined by the dataset. The file
+#' paths are relative paths for local files when the file srteam are downloaded
+#' or mounted. A common prefix will be removed from the file paths based on how
+#' data source was specified to create the dataset.
+#'
+#' @param dataset The Dataset object.
+#' @return A list of file paths.
+#' @export
+#' @md
+get_file_dataset_paths <- function(dataset) {
+  dataset$to_path()
```

Comment:
don't we need to convert this numpy.ndarray list to R list/dataframe object?

Reply:
It will be translated into an R array automatically - but I'll cast it as an R list.